code/modules/mob/living/carbon/xenomorph/hive_status.dm

```diff
@@ -91,6 +91,9 @@
 // Updates how many buried larva there are
 /datum/hive_status_ui/proc/update_burrowed_larva(send_update = TRUE)
 	burrowed_larva = assoc_hive.stored_larva
+	if(assoc_hive.no_burrowed)
+		if(assoc_hive.stored_larva > 0)
+			assoc_hive.stored_larva--
```

Comment:
this feels like the wrong way to do this; shouldn't this be done *on* the proc that adds stored larva?

Reply:
Yeah. But i gonna be really honest. Didn't found way, to work with main late join larva code. 
Cause i think, it need a little rework, if you wanna make it work with something hive_flag. Since now it just: Need to create larva from latejoin? +1 to all hives.

Reply:
OH ZANG IT ALSO PROBABLY GONNA REMOVE BURROWED LARVA'S FROM HIVE'. IF THERE LIKE... NORMAL HIVE WITH THAT FLAG AND CORRUPTED GET LARVA'S FROM MONKEY. Guh. Gonna check it